Formula/devcontainer.rb

```diff
@@ -0,0 +1,20 @@
+require "language/node"
+
+class Devcontainer < Formula
+  desc "Reference implementation for the Development Containers specification"
+  homepage "https://containers.dev"
+  url "https://registry.npmjs.org/@devcontainers/cli/-/cli-0.37.0.tgz"
+  sha256 "91faa34ab83dfd61a72361247816d434a026778806c795910f35343dae4ac591"
+  license "MIT"
+
+  depends_on "node"
+
+  def install
+    system "npm", "install", *Language::Node.std_npm_install_args(libexec)
+    bin.install_symlink Dir["#{libexec}/bin/*"]
+  end
+
+  test do
+    assert_match "0.37.0", shell_output("#{bin}/devcontainer --version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as you saw in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.